rdflib/namespace/__init__.py

```diff
@@ -564,25 +570,27 @@ def bind(
             raise KeyError("Prefixes may not contain spaces.")
 
         bound_namespace = self.store.namespace(prefix)
+
         # Check if the bound_namespace contains a URI
         # and if so convert it into a URIRef for comparison
         # This is to prevent duplicate namespaces with the
         # same URI
         if bound_namespace:
             bound_namespace = URIRef(bound_namespace)
-        if bound_namespace and bound_namespace != namespace:
 
-            if replace:
-                self.store.bind(prefix, namespace)
-                insert_trie(self.__trie, str(namespace))
-                return
+            self.store.bind(prefix, namespace, replace=replace)
+            insert_trie(self.__trie, str(namespace))
 
+            return
+
+        if bound_namespace and bound_namespace != namespace:
```

Comment:
This branch is unreachable, because if bound_namespace is truthish then the previous if will have matched and function would have returned.

I think @gjhiggins  may have noted this